subsys/pm/device.c

```diff
@@ -101,6 +101,57 @@ int pm_device_action_run(const struct device *dev,
 	return 0;
 }
 
+int pm_device_power_domain_add(const struct device *dev,
+			       const struct device *domain)
+{
+#ifdef CONFIG_PM_DEVICE_POWER_DOMAIN
+	device_handle_t *rv = (device_handle_t *)domain->handles;
```

Comment:
Isn't this line discarding the `const` qualifier from the handles?
https://github.com/zephyrproject-rtos/zephyr/blob/a5b33f89b729d5803e3a6879d89eaef3b98a5ebd/include/device.h#L468
AFAIK the handles currently live in ROM, so I'm not sure how this can work

Reply:
It is resized in compile time, so the size is pre-allocated.

Reply:
You're completely right, I've tested in the integration test (that is using native posix) and didn't catch it.

Reply:
So, I just did some small changes, when this option is enabled, the array is put in the ram instead of the rom. It is indeed memory cost, but it is a price to pay for it. If this option  is not set, anything changes and should not be any impact. A future change may put just an additional array in RAM and keep the ones know at build time in ROM. This change though, will not require any API change and will be done entirely internally.